src/sentry/eventstream/kafka/backend.py

```diff
@@ -210,14 +214,26 @@ def handle_shutdown_request(signum: int, frame: Any) -> None:
             i = i + 1
             owned_partition_offsets[key] = message.offset() + 1
 
-            with metrics.timer("eventstream.duration", instance="get_task_kwargs_for_message"):
-                task_kwargs = get_task_kwargs_for_message(message.value())
+            use_kafka_headers = options.get("post-process-forwarder:kafka-headers")
 
-            if task_kwargs is not None:
-                with metrics.timer(
-                    "eventstream.duration", instance="dispatch_post_process_group_task"
-                ):
```

Comment:
Yeah this may be worth looking into later. I'm not aware of anything we're using it for (but hard to be sure), and I believe we're just applying the default sample rate of 1 (no sampling).